docker-compose.yml

```diff
@@ -32,6 +31,49 @@ services:
     depends_on:
       admin_backend_db:
         condition: service_healthy
-
+  pgadmin:
+    image: dpage/pgadmin4
+    container_name: pgadmin4
+    restart: always
+    env_file:
+      - .env
+    ports:
+      - 8000:80
+    # This has to have world & group access denied to work
+    entrypoint: /bin/sh -c "chmod 600 /pgpass; /entrypoint.sh;"
+    configs:
+      - source: servers.json
+        target: /pgadmin4/servers.json
+      - source: pgpass
+        target: /pgpass
+    environment:
+      # This deavctivates the login screen
+      PGADMIN_CONFIG_SERVER_MODE: "False"
+      PGADMIN_CONFIG_MASTER_PASSWORD_REQUIRED: "False"
+    volumes:
+      - pgadmin-data:/var/lib/pgadmin
+    # This is just nice to know as if it fails, it fails quietly
+    healthcheck:
+      test: [CMD, wget, -O, "-", http://localhost:80/misc/ping]
+configs:
```

Comment:
I'd love some more info on this `configs` section as I haven't come across it before! Wha's the benefit of this over extending the out of the box pgadmin4 image and configuring inside that extended image?

Reply:
When you say extending the pg4admin image - what do you mean? Creating a new `Containerfile` for it?

My thinking on this was that pgAdmin makes these options configurable via files that exist on the filesystem at startup. A way that's recommended by Docker is to use [configs](https://docs.docker.com/reference/compose-file/services/#configs) ([here's the top level element](https://docs.docker.com/reference/compose-file/configs/)).

I was you could specify it directly on the specific `service` - but their API isn't shaped that way.

Not sure I've answered the question?

Reply:
Thanks for sharing - I hadn't come across this before! Good to know :)